cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -343,18 +350,18 @@ instance ToJSON ProtocolParameters where
       , "collateralPercentage"   .= protocolParamCollateralPercent
       , "maxCollateralInputs"    .= protocolParamMaxCollateralInputs
       ]
-    where
-      -- Rationals and JSON are an awkward mix. We cannot convert rationals
-      -- like @1/3@ to JSON numbers. But _most_ of the numbers we want to use
-      -- in practice have simple decimal representations. Our solution here is
-      -- to use simple decimal representations where we can and representation
-      -- in a @{"numerator": 1, "denominator": 3}@ style otherwise.
-      --
-      toRationalJSON :: Rational -> Aeson.Value
-      toRationalJSON r =
-        case Scientific.fromRationalRepetend (Just 5) r of
-          Right (s, Nothing) -> toJSON s
-          _                  -> toJSON r
+
+-- Rationals and JSON are an awkward mix. We cannot convert rationals
+-- like @1/3@ to JSON numbers. But _most_ of the numbers we want to use
+-- in practice have simple decimal representations. Our solution here is
+-- to use simple decimal representations where we can and representation
+-- in a @{"numerator": 1, "denominator": 3}@ style otherwise.
+--
+toRationalJSON :: Rational -> Aeson.Value
+toRationalJSON r =
+  case Scientific.fromRationalRepetend (Just 5) r of
+    Right (s, Nothing) -> toJSON s
+    _                  -> toJSON r
```

Comment:
Yes. For whatever reason, although the `Rational` instance for `FromJSON` accepts both number and object style, the `ToJSON` instance never produces numbers, only object style:
```
instance (ToJSON a, Integral a) => ToJSON (Ratio a) where
    toJSON r = object [ "numerator"   .= numerator   r
                      , "denominator" .= denominator r
                      ]
    {-# INLINE toJSON #-}

    toEncoding r = E.pairs $
        "numerator" .= numerator r <>
        "denominator" .= denominator r
    {-# INLINE toEncoding #-}
```
So we do need this `toRationalJSON` workaround.

Perhaps that is worth proposing upstream as a PR to do the above trick for the instance provided by the library.

Reply:
And also:

```
λ> encode (1 :: Rational)
"{\"numerator\":1,\"denominator\":1}"
```

Reply:
Until and if such change gets merged upstream, I'd also suggest importing `toRationalJSON` from `Cardano.Api.Orphans` instead of redefining another copy again here.

Reply:
I hesitate mainly because I can't use it unless I export it from `Cardano.Api` anyway, in which case it becomes part of our API which I don't want especially because we want to push it upstream in a way that doesn't need an extra function.

Reply:
I did not realize `Cardano.Api.Orphans` is an exposed module. That is a very strange thing to do considering that anyone who imports `Cardano.Api` will get all those orphan instances into scope anyways (all instances are imported through modules transitively). I would suggest moving that module into `other-modules` in the cabal file

Reply:
Actually, my mistake.  I thought I was using `toRationalJSON` from outside `cardano-api`, but I don't.  But it still is a bit strange to export functions from a module that called Orphans.  Not a hard objection though.

Reply:
I will move it to a new module `Cardano.Api.Json`.  Seems the cleanest.